src/twister2/report/test_results_plugin.py

```diff
@@ -139,7 +139,10 @@ def _get_outcome(self, report: pytest.TestReport) -> str | None:  # type: ignore
             else:
                 return Status.SKIPPED
         elif report.passed and report.when == 'call':
-            return Status.PASSED
+            if hasattr(report, 'wasxfail'):
+                return Status.XPASSED
```

Comment:
In previous checks if `wasxfail` returned XFAILED status and here the same check returns XPASSED. Is this correct? can you please explain this part?

Reply:
This is the case where test is marked as `pytest.mark.xfail(reason='expected to fail')` but the test passed (but should failed) so we want to have an information that the test `unexpected` passed (this is the status that pytest returns).